### PR TITLE
Prepend user-specified PATH to existing PATH

### PR DESCRIPTION
**Pull request by jheaff1**
If a user specifies a PATH value as part of an `env` attribute, the
value will be prepended to the existing PATH.

An example requirement for this change is that the MSVC build of
OpenSSL requires that the Netwide Assembler (NASM) must be on the
PATH.

### CODE REVIEW

**Review by UebelAndre — APPROVED** [review]
Thanks! This seems good to me but I'd love to get @jsharpe to take a look too!

File: foreign_cc/private/framework.bzl [hunk 1]
```diff
@@ -103,7 +103,8 @@ CC_EXTERNAL_RULE_ATTRIBUTES = {
             "`$(execpath)` macros may be used to point at files which are listed as `data`, `deps`, or `build_data`, " +
             "but unlike with other rules, these will be replaced with absolute paths to those files, " +
             "because the build does not run in the exec root. " +
-            "No other macros are supported."
+            "No other macros are supported." +
+            "PATH entries will be prepended to the existing PATH."
```
Comment (jsharpe) [inline foreign_cc/private/framework.bzl:107]: nit:
```suggestion
            "`PATH` entries will be prepended to the existing `PATH`."
```

**Review by jsharpe — COMMENTED** [review]
One small nit in the doc string.

Only comment I have is should we actually be prepending to all keys that exist in the `cc_env` dict and not just `PATH` e.g. `LIBRARY_PATH`, `CPATH`, `LD_LIBRARY_PATH`, `DYLD_LIBRARY_PATH` are all possible other env variables that we might want to preserve? I'd assume that anything that gets set in `cc_env` is required for the proper functioning of the configured `cc_toolchain` and so we should be preserving it.

**Comment by jheaff1** [comment]
> One small nit in the doc string.
> 
> Only comment I have is should we actually be prepending to all keys that exist in the `cc_env` dict and not just `PATH` e.g. `LIBRARY_PATH`, `CPATH`, `LD_LIBRARY_PATH`, `DYLD_LIBRARY_PATH` are all possible other env variables that we might want to preserve? I'd assume that anything that gets set in `cc_env` is required for the proper functioning of the configured `cc_toolchain` and so we should be preserving it.

Actioned

File: foreign_cc/private/framework.bzl (hunk 1 above)
Comment (jheaff1) [reply foreign_cc/private/framework.bzl:107]: Actioned

**Comment by jheaff1** [comment]
Can this be merged now? 😁

**Review by jsharpe — APPROVED** [review]

**Comment by jsharpe** [comment]
> Can this be merged now? 😁

Done. Thanks!

**Comment by jheaff1** [comment]
Thank you!

**Pull request merged**